src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h

```diff
@@ -120,13 +120,15 @@ class GenericThreadStackManagerImpl_OpenThread
     struct SrpClient
     {
         static constexpr uint8_t kServiceId           = 0x5d;
-        static constexpr uint8_t kMaxServicesNumber   = 3;
-        static constexpr uint8_t kMaxInstanceNameSize = 64;
-        static constexpr uint8_t kMaxNameSize         = 16;
+        static constexpr uint8_t kMaxServicesNumber   = CHIP_DEVICE_CONFIG_THREAD_SRP_MAX_SERVICES;
+        static constexpr uint8_t kMaxInstanceNameSize = chip::Mdns::kMdnsNameMaxSize;
+        static constexpr uint8_t kMaxNameSize         = chip::Mdns::kMdnsTypeMaxSize + chip::Mdns::kMdnsProtocolTextMaxSize + 1;
         static constexpr uint8_t kMaxHostNameSize     = 32;
-        static constexpr uint8_t kMaxTxtEntriesNumber = 4;
-        static constexpr uint8_t kMaxTxtValueSize     = 255;
-        static constexpr uint8_t kMaxTxtKeySize       = 16;
+        // Thread only supports operational discovery currently, increase when commissionable discovery would get added
+        // see GH issue:#5351
+        static constexpr uint8_t kMaxTxtEntriesNumber = 2;     // Based on operational discovery - 2 records required
```

Comment:
Can we make these constants like above?

Reply:
I'll see to extend the defines in mdns.h where these fields are being used and populated

Reply:
@woody-apple 
I added the defines now in the 2 types of advertising classes in the Advertiser.h header. That looked like the closest place to put these constants should anything get altered to the buffer sizes.

Created a TODO for the commissionable advertising bit as that didn't look to adhere to spec yet - only 4 of the 8 possible txt records are implemented.